ops/charm.py

```diff
@@ -90,92 +111,150 @@ def fail(self, message: str = '') -> None:
 
 
 class InstallEvent(HookEvent):
-    """Represents the `install` hook from Juju."""
+    """Event triggered when a charm is installed.
+
+    This event is triggered at the beginning of a charm's
+    lifecycle. Any associated callback method should be used to
+    perform one-time setup operations, such as installing prerequisite
+    software.
+    """
 
 
 class StartEvent(HookEvent):
-    """Represents the `start` hook from Juju."""
+    """Event triggered immediately after first configuation change.
+
+    This event is triggered immediately after the first :class:
+    `ConfigChangedEvent` event. Callback methods bound to the event
+    should be used to ensure that the charm’s software is in a running
+    state. Note that the charm’s software should be configured so as
+    to persist in this state through reboots without further
+    intervention on Juju’s part.
+    """
 
 
 class StopEvent(HookEvent):
-    """Represents the `stop` hook from Juju."""
+    """Event triggered when a charm is shut down.
+
+    This event is triggered when an application's removal is requested
+    by the client. The event fires immediately before the end of the
+    unit’s destruction sequence. Callback methods bound to this event,
+    should be used to ensure that the charm’s software is not running,
+    and will not start again on reboot.
+    """
 
 
 class RemoveEvent(HookEvent):
-    """Represents the `remove` hook from Juju. """
+    """Represents the `remove` hook from Juju.
+    """
 
 
 class ConfigChangedEvent(HookEvent):
-    """Represents the `config-changed` hook from Juju."""
+    """Event triggered when a configuration change is requested.
+
+    This event fires in several different situations.
+
+    - immediately after `install` event.
+    - immediately after `upgrade_charm` event.
+    - at least once when the unit agent is restarted (but, if the unit
+      is in an error state, it won’t be run until after the error
+      state is cleared).
+    - after changing charm configuration using the GUI or command line
+      interface
+
```

Comment:
Thank you. I will add a line stating that network information change results in config change. Appreciate you point about future changes. If you rather leave this out for now let me know, otherwise IMHO accuracy is the best way to help our charmers. I will gladly help update these docs in the future when things change